nep-Y: add missing return error codes

AnnaShaleva opened the pull request:
Based on the
https://docs.google.com/spreadsheets/d/1VShwR1qdVeE7J4gGUBzSPnjAIWkhXb-VT9yvf9cau6I/edit#gid=0 and https://github.com/nspcc-dev/neo-go/pull/3063.

I've added comments to the table wrt the rest of issues specified in the https://docs.google.com/spreadsheets/d/1VShwR1qdVeE7J4gGUBzSPnjAIWkhXb-VT9yvf9cau6I/edit#gid=0.

AnnaShaleva commented on nep-Y.mediawiki:
```diff
@@ -288,17 +288,17 @@ Neo-specific codes are documented here, generic JSON-RPC errors like -32602 or
 | sendrawtransaction  || -500, -501, -502, -503, -504, -505, -506, -507, -508, -509, -510, -511
 |-
-| submitblock         || -500, -501, -508, -509
+| submitblock         || -500, -501, -502, -503, -504, -505, -506, -507, -508, -509, -510, -511
```
This one is NeoGo feature (we create block-level mempool while verifying in-block transactions), but still, these errors can be returned by the NeoGo node on the protocol level. Thus, I consider that we either have to add them to the protocol or document them in NeoGo as protocol extensions. I prefer the first, so that the protocol users will be aware of the possibility to get the extended set of errors.

roman-khimov replied:
Well, this needs to be committed with an explanation like "blocks contain transactions, therefore any transaction-related error codes can be applicable too".

AnnaShaleva replied:
Fixed.

AnnaShaleva commented:
@roman-khimov, I'm not able to request your review here, thus I'm leaving this comment.

roman-khimov commented on nep-Y.mediawiki:
```diff
@@ -210,11 +210,11 @@ Neo-specific codes are documented here, generic JSON-RPC errors like -32602 or
+| getblock            || -101, -109
 |-
-| getblockhash        || -109
+| getblockhash        || -101, -109
```
Why -101 here? It just resolves height to hash, so invalid params or wrong height are expected, but -101 is not.

AnnaShaleva replied:
Removed.

roman-khimov commented on nep-Y.mediawiki:
```diff
@@ -210,11 +210,11 @@ Neo-specific codes are documented here, generic JSON-RPC errors like -32602 or
+| getblockhash        || -101, -109
 |-
-| getblockheader      || -101
+| getblockheader      || -101, -109
```
getblock/getblockheader are OK (commit them with "these can autoresolve height to hash, so height-related error code is applicable as well")

AnnaShaleva replied:
Fixed.

roman-khimov commented on nep-Y.mediawiki:
```diff
@@ -288,17 +288,17 @@ Neo-specific codes are documented here, generic JSON-RPC errors like -32602 or
+| submitblock         || -500, -501, -502, -503, -504, -505, -506, -507, -508, -509, -510, -511
 |-
-| submitoracleresponse|| -602, -603, -604, -605
+| submitoracleresponse|| -508, -602, -603, -604, -605
```
Makes sense.

AnnaShaleva replied:
Moved to a separate commit.

roman-khimov commented on nep-Y.mediawiki:
```diff
@@ -288,17 +288,17 @@ Neo-specific codes are documented here, generic JSON-RPC errors like -32602 or
 | validateaddress     ||
 |-
-| verifyproof         || -607
+| verifyproof         || -606, -607
```
And this one too. I'd just have them logically separated in commits, each modification has its own reason behind and we need to explain the semantics of these changes.

AnnaShaleva replied:
Fixed.

AnnaShaleva committed:
NEP-Y: extend list of `getblock` and `getblockheader` return error codes

These can autoresolve height to hash, so height-related error code
is applicable as well.

Signed-off-by: Anna Shaleva <shaleva.ann@nspcc.ru>

AnnaShaleva committed:
NEP-Y: extend list of `submitoracleresponse` return error codes

Oracle response can have invalid signature.

Signed-off-by: Anna Shaleva <shaleva.ann@nspcc.ru>

AnnaShaleva committed:
NEP-Y: extend list of `submitblock` return error codes

Blocks contain transactions, therefore any transaction-related
error codes can be applicable too.

Signed-off-by: Anna Shaleva <shaleva.ann@nspcc.ru>

AnnaShaleva committed:
NEP-Y: extend list of `verifyproof` return error codes

Proof verification is relevant only for the nodes that store not only
the latest chain state.

Signed-off-by: Anna Shaleva <shaleva.ann@nspcc.ru>
